kafe2/fit/_base/cost.py

```diff
@@ -290,57 +225,94 @@ def on_no_errors(self):
             self._no_errors_warning_printed = True
 
 
-class CostFunctionBase_Chi2(CostFunctionBase):
+class CostFunction_Chi2(CostFunction):
     def __init__(self, errors_to_use='covariance', fallback_on_singular=True):
         """
         Base class for built-in least-squares cost function.
 
         :param errors_to_use: which errors to use when calculating :math:`\chi^2`
         :type errors_to_use: ``'covariance'``, ``'pointwise'`` or ``None``
```

Comment:
For better use of intersphinx and better type hinting with advanced Editors like PyCharm I would use:
```suggestion
        :param errors_to_use: Which errors to use when calculating :math:`\chi^2`.
                              Either ``'covariance'``, ``'pointwise'`` or ``None``.
        :type errors_to_use: str or None
```